packages/react/src/internal/components/TextInputWrapper.tsx

```diff
@@ -117,108 +56,208 @@ export const TextInputBaseWrapper = styled.span<StyledBaseWrapperProps>`
   textarea,
   select {
     &::placeholder {
-      color: var(---control-fgColor-placeholder, ${get('colors.fg.muted')});
+      color: var(---control-fgColor-placeholder, var(--fgColor-muted));
     }
   }
 
-  ${props => renderFocusStyles(Boolean(props.hasTrailingAction), Boolean(props.isInputFocused))}
+  &:where([data-trailing-action][data-focused]),
```

Comment:
Is `data-focused` something where we could use `:focus` or `:focus-visible` or is this something that's been providing through props or determined at runtime? Wasn't sure so wanted to double check 👀

Reply:
It felt prudent to keep the logic matching to the existing styling. The contents of the wrapper could change and didn't want to rely on its state, but what is being passed in by the prop